airbyte-webapp/src/core/form/schemaToYup.ts

```diff
@@ -0,0 +1,208 @@
+import { JSONSchema7 } from "json-schema";
+import * as yup from "yup";
+
+import { FormBlock, FormGroupItem, FormObjectArrayItem, FormConditionItem } from "core/form/types";
+import { isDefined } from "utils/common";
+
+import { FormBuildError } from "./FormBuildError";
+
+/**
+ * Returns yup.schema for validation
+ *
+ * This method builds yup schema based on jsonSchema ${@link JSONSchema7} and the derived ${@link FormBlock}.
+ * Every property is walked through recursively in case it is condition | object | array.
+ *
+ * @param jsonSchema
+ * @param formField The corresponding {@link FormBlock} to the given schema
+ * @param parentSchema used in recursive schema building as required fields can be described in parentSchema
+ * @param propertyKey used in recursive schema building for building path for uiConfig
+ * @param propertyPath constructs path of property
+ */
+export const buildYupFormForJsonSchema = (
+  jsonSchema: JSONSchema7,
+  formField: FormBlock,
+  parentSchema?: JSONSchema7,
+  propertyKey?: string,
+  propertyPath: string | undefined = propertyKey
+): yup.AnySchema => {
+  let schema:
+    | yup.NumberSchema
+    | yup.StringSchema
+    | yup.AnyObjectSchema
+    | yup.ArraySchema<yup.AnySchema>
+    | yup.BooleanSchema
+    | null = null;
+
+  if (jsonSchema.oneOf && propertyPath) {
```

Comment:
This whole if block took me a few tries to understand as it is fairly complex. It would be great if we could simplify it a bit but I'm not immediately seeing a way to do that, and I don't think that should necessarily block merging

Reply:
Yeah, that's pretty dense code. I added an example to the comment to explain what's going on.

Reply:
The examples are super helpful, thanks!